walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/requirements/constraints/GroupConstraint.kt

```diff
@@ -31,4 +34,27 @@ class GroupConstraint(
             }
         }
     }
+
+    override fun matches(verifiedId: VerifiedId) {
+        val validationExceptions = mutableListOf<String>()
+        constraints.forEach { constraint ->
+            try {
+                constraint.matches(verifiedId)
+            } catch (exception: WalletLibraryException) {
+                exception.message?.let { validationExceptions.add(it) }
+            }
+        }
+
+        when (constraintOperator) {
+            GroupConstraintOperator.ANY -> {
+                if (constraints.size == validationExceptions.size)
+                    throw NoMatchForAnyConstraintsException("None of the constraints match.")
```

Comment:
Is there a way to surface which constraints do not match?

Reply:
We could add the name of constraint class as part of exception message.

Reply:
Added it.